### DISCUSSION TITLE
编程学习问题

### DISCUSSION THREAD

**Discussion by mengzhi73** [MaaAssistantArknights/MaaAssistantArknights#8855] [Q&A] [answered]
rt，maa这种脚本程序需要具备哪些编程知识才能开发，孩子大一计算机专业的，目前只学过c++，python，linux的一些基础知识，有没有大佬给个学习路线啊

**Comment by ABA2396** [comment, marked as answer]
maa core 使用 c++ 编写，wpf ui 使用 c# 编写

maa 基于图像识别技术，因此需要对 OpenCV 等图像处理库有所了解，并理解一些基本的图像处理技术，如滤波，边缘检测，模式匹配等

主要的逻辑是在代码和 json 任务链中实现的，需要对 json 格式有了解，任务流程协议在 [这里](https://maa.plus/docs/zh-cn/protocol/task-schema.html) ，其他的可以之后再看

UI 部分需要了解 .NET 框架，并且能够使用 WPF 进行用户界面的开发

做为软件开发的基础，需要理解软件开发的生命周期，如需求分析，设计，编码，测试和维护等工作。当然前提是能看懂代码，maa core 使用的是 c++20 标准，可能和你学校学的差距比较大

如果想要贡献代码，需要对github和git有一定的了解，这部分可以看看 [纯网页端PR教程](https://maa.plus/docs/zh-cn/develop/pr-tutorial.html) 或者 [github-pull-request-流程简述](https://maa.plus/docs/zh-cn/develop/development.html) ，当然这些只是最基础的，如果想要深入请自行查找资料

python 涉及的不多，主要是做为下载依赖和小工具的编写使用，https://github.com/MaaAssistantArknights/MaaAssistantArknights/tree/dev/tools 里很多都是用 py 写的，需要有一定的了解

部分地方用到了机器学习和深度学习，主要还是决策树和图像识别模型，模型训练部分有兴趣的话可以看看 https://github.com/MaaAssistantArknights/MaaAI

开始开发前先看 [开始开发](https://maa.plus/docs/zh-cn/develop/)

**Comment by MistEO** [comment]
想自己开发一个其他游戏的脚本可以试试 [MaaFramework](https://github.com/MaaXYZ/MaaFramework/blob/main/docs/zh_cn/1.1-%E5%BF%AB%E9%80%9F%E5%BC%80%E5%A7%8B.md)

MaaFramework 是基于图像识别技术、运用 MAA 开发经验去芜存菁、完全重写的新一代自动化黑盒测试框架。

低代码的同时仍拥有高扩展性，旨在打造一款丰富、领先、且实用的开源库，助力开发者轻松编写出更好的黑盒测试程序，并推广普及。
